8.x/theme-development/client-side-cart-module.md

```diff
@@ -0,0 +1,665 @@
+## Client Side cart module
+
+[[toc]]
+
+Larammerce base theme has a build-in JS module named **LocalCartService** to manage the client-side cart in the customer's browser.
+This document reviews the life cycle and progress of this module.
+
+
+**LocalCartService** is placed in the `resources/assets/js/define/local_cart_service.js` file consisting of 437 lines of code. It is based on require js module. To read more about it, refer to this address:[requirejs](https://www.requirejs.org/).
+
+First, let's check how **LocalCartService** manages the shopping cart:
+
+The image below is one of the websites created with Larammerce, and it can be seen that if the user is not logged in, the module manages the shopping cart, and if the user selects a product, it is added to the shopping cart.
+
+As you should know, when the user is not logged in, the shopping cart data is stored as JSON in the cookie and the data is stored on the server if the user is logged in.
+
+![LocalCartService.png](/LocalCartService.png)
+
+To see how the products are stored in the shopping cart, copy the value of local_cart_v11, and after decoding, the following value will be displayed:
+
+```json
+{
+	"673": {
+		"count": 10
+	},
+	"676": {
+		"count": 1
+	}
+}
+```
+There are two products in the shopping cart with the keys 673 and 676, which are the product IDs, and the count value is the number of products stored in the shopping cart.
+
+After logging in, server side requests are checked to see how the data is stored on the server:
+
+![LocalCartService-update-count.png](/LocalCartService-update-count.png)
+
+After increasing the number of products, a request is sent to the server. In the request address, it can be seen that the number of products in the cart has been updated, and the number of products has increased to 11.
+
+
+
+Now which ajax request is called in the shopping cart process is checked:
+
+![LocalCartService-shopping-cart-process.png](/LocalCartService-shopping-cart-process.png)
+
+In the above image, it can be seen that there were ten products in the shopping cart at first. After reducing the number of products, This request: `URL: https://retopi.com/customer/cart/update-count/673?count=7` is sent to the server, which means that the number of products in the shopping cart will be updated.
```

Comment:
Please don't mention the hostname of the website.
Just use the /customer/cart/update-count/.... to show the web service.
Finally, please mention that: calling the web service would be led the data to be saved on the server-side (aka database).